app/models/digital_object.rb

```diff
@@ -13,6 +13,15 @@ def initialize(label:, href:)
 
   def self.from_json(json)
     object_data = JSON.parse(json)
-    new(label: object_data['label'], href: "https://purl.stanford.edu/#{object_data['href']}")
+    new(label: object_data['label'], href: normalize_href(object_data['href']))
+  end
+
+  # Make a usable Purl URL from whatever happens to be in
+  # the DigitalObject href in the sample data.
+  # Ideally, this value would be consistent, but it is not currently.
+  def self.normalize_href(href)
+    return href.gsub('http://', 'https://') if href.match?(%r{https?://})
+
+    "https://purl.stanford.edu/#{href}"
```

Comment:
Same kind of question here.. Is the any benefit to validating that it's a druid-shaped string before making it a PURL?